packages/block-editor/src/components/block-list-appender/index.native.js

```diff
@@ -22,12 +23,24 @@ function BlockListAppender( {
 	canInsertDefaultBlock,
 	isLocked,
 	renderAppender: CustomAppender,
+	appender,
 	showSeparator,
 } ) {
-	if ( isLocked ) {
+	if ( undefined !== CustomAppender ) {
+		deprecated( 'renderAppender prop of the BlockListAppender component', {
+			since: '5.7',
+			alternative: 'appender prop of the BlockListAppender component',
+		} );
```

Comment:
I think most implementers will specify this prop on the `InnerBlocks` component or `useInnerBlocksProps` hook. They probably won't be aware of `BlockListAppender`.

It might be easiest to omit the component name from the message, or change the wording to 'renderAppender prop of InnerBlocks component or useInnerBlocksProps hook'.

Reply:
Or alternatively could list all the components that this effects, e.g.
''renderAppender prop of the BlockListAppender and InnerBlocks components, and the useInnerBlocksProps hook'

Reply:
Makes sense! I decided to just omit the component, otherwise, it would demand the message maintenance depending on the components hierarchy changes. ;)

Reply:
Also added a changelog entry here: https://github.com/WordPress/gutenberg/pull/29911/commits/44f6fa5ee4e98f13a15a73e2bff52e20214ee395